### PR TITLE
Use Cython to build extension modules

### LINKED ISSUE #333: Don't put Cythonized .c files in source control, but do ship in sdist
Currently we check-in .c files produced by Cython to the source tree alongside the .pyx files so that people building the source don't need to have Cython installed.  This is awkward from the developer's perspective, however, and can result in noisy deltas.

Following discussion in #325 the the proposal is that we will only check in the .pyx files into source control, but we will ship the .c files as part of the sdist source distributions.  This change will mean that people wishing to work from non-released versions will need to have Cython installed (as will the CI environment), but people wanting to build a release from source won't need it.  Having Cython available is not as unreasonable a requirement as it was several years ago.

### PR DESCRIPTION
**Pull request by jvkersch**
Removes the autogenerated .c files and relies on Cython being present at build time to build the extension modules.

Supersedes #459

Fixes #333

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -5,6 +5,7 @@ include chaco/tools/toolbars/images*.png
 include chaco/tools/toolbars/images*.svg
 include chaco/tools/toolbars/images*.txt
 include chaco/layers/data/*.svg
+recursive-include chaco *.pyx
```
Comment (rahulporuri) [inline MANIFEST.in]: Am i missing something or should we be including the `cntr.c` file?
#459 was doing this.

**Comment by rahulporuri** [comment]
can you also update the PR description with a ref to the issue that this PR addresses?

File: MANIFEST.in (hunk 1 above)
Comment (jvkersch) [reply MANIFEST.in]: C files that are listed in extension modules are included by default (https://packaging.python.org/guides/using-manifest-in/). I missed that in #459.

**Comment by jvkersch** [comment]
Done and done. Thanks @rahulporuri for the quick review. Happy to enter a new decade of Python hacking with you :tada: :fireworks:

**Pull request merged**